CRM/Core/Permission/Drupal8.php

```diff
@@ -93,11 +93,12 @@ public function permissionEmails($permissionName) {
     if (isset($_cache[$permissionName])) {
       return $_cache[$permissionName];
     }
-
+    $roles = \Drupal\user\Entity\Role::loadMultiple();
+    unset($roles[\Drupal\user\RoleInterface::ANONYMOUS_ID]);
     $role_ids = array_map(
       function (\Drupal\user\RoleInterface $role) {
         return $role->id();
-      }, user_roles(TRUE, $permissionName)
+      }, array_filter($roles, fn(\Drupal\user\RoleInterface $role) => $role->hasPermission($permissionName))
```

Comment:
I checked and this `fn` syntax still works in php 7.4. 👍 

I find I have to watch my effin' syntax or else the effin' computer goes api-shift...